### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by vext01**
Here's a handful of small changes that came out of my experimentation over the past few days.

For reviewing, it's probably best to look at individual commits.

Cheers

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -125,6 +125,7 @@ pub enum Terminator {
     Call {
         operand: CallOperand,
         cleanup_bb: Option<BasicBlockIndex>,
+        ret_bb: Option<BasicBlockIndex>,
```
Comment (ltratt) [inline src/lib.rs]: How does this change work in isolation? Wouldn't it have either a) not compiled before this b) not compile now?

Comment (vext01) [reply src/lib.rs]: I'll update the commit message to explain.

**Comment by vext01** [comment]
Give this a shot :)

File: src/lib.rs [hunk 2]
```diff
@@ -85,7 +85,7 @@ impl BasicBlock {
     }
 }
 
-/// A MIR statement.
+/// A Statement.
```
Comment (ltratt) [inline src/lib.rs]: Should be lower-case "statement"? [As a minor issue, I'm also not sure that this comment is worth the line it takes up!]

Comment (vext01) [reply src/lib.rs]: I tend to agree that this comment can go.

[This is a habit I've picked up from rustc, where it sometimes insists that every item is documented]

**Comment by vext01** [comment]
I've reviewed all of the comments in our types.rs and removed any which don't really add any value.

Let me know if you disagree with any of those and I can bring them back, otherwise I'd like to squash this last commit.

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
Here we go.

**Pull request merged**